### PR TITLE
Use TCP health check of HAProxy if app's health check protocol is "TCP"

### PR DESCRIPTION
**Pull request by fpgeek**
I think the TCP health check of HAProxy should be used if application's mode is "http" but health check is "tcp".
Because an application depends on it's health check.
How do you think about it?

### CODE REVIEW

File: marathon_lb.py [hunk 1]
```diff
@@ -805,7 +808,7 @@ def config(apps, groups, bind_http_https, ssl_certs, templater):
             healthCheckOptions = None
             if app.healthCheck:
                 server_health_check_options = None
-                if app.mode == 'tcp':
+                if app.mode == 'tcp' or app.healthCheck['protocol'] == 'TCP':
```
Comment (brndnmtthws) [inline marathon_lb.py]: Maybe do a `.lower()` on these strings before the comparison.

File: marathon_lb.py [hunk 2]
```diff
@@ -756,7 +759,7 @@ def config(apps, groups, bind_http_https, ssl_certs, templater):
 
         if app.healthCheck:
             health_check_options = None
-            if app.mode == 'tcp':
+            if app.mode == 'tcp' or app.healthCheck['protocol'] == 'TCP':
```
Comment (brndnmtthws) [inline marathon_lb.py]: `.lower()` here too.

File: tests/test_marathon_lb.py [hunk 3]
```diff
@@ -220,7 +220,7 @@ def test_config_healthcheck_command(self):
 backend nginx_10000
   balance roundrobin
   mode tcp
-  server 1_1_1_1_1024 1.1.1.1:1024
+  server 1_1_1_1_1024 1.1.1.1:1024 check inter 2s fall 11
```
Comment (brndnmtthws) [inline tests/test_marathon_lb.py]: Hmm, why did this change?

**Comment by brndnmtthws** [comment]
Seems reasonable. Should be good to go once you address the comments above.

File: marathon_lb.py (hunk 1 above)
Comment (fpgeek) [inline marathon_lb.py]: Health check protocol of Marathon allows only "TCP". "tcp" is not allowed.

File: marathon_lb.py (hunk 2 above)
Comment (fpgeek) [inline marathon_lb.py]: "tcp" is not allowed.

File: tests/test_marathon_lb.py (hunk 3 above)
Comment (fpgeek) [inline tests/test_marathon_lb.py]: Default value of "HAPROXY_BACKEND_SERVER_TCP_HEALTHCHECK_OPTIONS" has changed.

Comment (brndnmtthws) [inline tests/test_marathon_lb.py]: Makes sense, I missed that.

File: marathon_lb.py (hunk 1 above)
Comment (brndnmtthws) [inline marathon_lb.py]: I think you're right...although it's confusing that Marathon uses lowercase for the portmappings, but uppercase for the health checks.

**Comment by brndnmtthws** [comment]
Thanks!

**Pull request merged**